tests/integration_test.rs

```diff
@@ -108,7 +111,9 @@ fn test_integration() {
         ..Default::default()
     };
     let (_stop_signal_send, stop_signal_receive) = bounded(1);
-    let mut p = Profiler::new(profiler_config, stop_signal_receive);
+    let metadata_provider: ThreadSafeGlobalMetadataProvider =
```

Comment:
nit: No need to specify the type
```suggestion
    let metadata_provider =
```